migrations/emailmonitor/oxa_email_monitor.py

```diff
@@ -0,0 +1,215 @@
+        if cursor.rowcount <= 0: 
+		    # Find user_id from email 
+            cursor.execute("SELECT id,date_joined from edxapp.auth_user WHERE email='"+email+"'")
+            user_id = 0
+			# Here we are converting time to PST time by subtracting 7, Normally in log files it is UTC time. But in MySQL it is PST time. So we map them to eachother as PST time
+            date_joined = dt_obj + timedelta(hours=7)
+            date_failed = date_joined
+            #Find the activation ket 
+			activation_key = "Not Found!!!"
+            # Find activation key from user_id
+			if cursor.rowcount > 0:
+                result = cursor.fetchall()         
+                user_id = result[0][0]
+                date_joined = result[0][1] 
+                cursor.execute("SELECT activation_key FROM edxapp.auth_registration WHERE user_id="+str(user_id)) 
+                if cursor.rowcount > 0:
+                    result = cursor.fetchall()
+                    activation_key = result[0][0]
+            
+			# Insert this new activation email failure to oxa.oxa_activationfailed table and update activation_failed statistics for corresponding day in oxa.oxa_activationsummary table
+            cursor.execute("INSERT INTO oxa.oxa_activationfailed (xyear,xmonth,xday,user_id,email,activation_key,date_joined,date_failed,hostvmip) VALUES("+y+","+mo+","+d+","+str(user_id)+",'"+email+"','"+activation_key+"','"+str(date_joined)+"','"+str(date_failed)+"','"+ip+"')")
+            cursor.execute("UPDATE oxa.oxa_activationsummary SET failed=failed+1 WHERE  xyear="+y+" and xmonth="+mo+" and xday="+d)    
+             
+        
+
+    
+# Fetch the error log files for the specified IP VMs and process them in order to determine activation email failures
+def fetch_and_grep_log_files():
+    # Remove if exists and create the folder for fethcing error log files from edxapp VMs
+	os.system('rm -fr /tmp/oxa_log_files')
+    os.system('mkdir /tmp/oxa_log_files')
+
+	# For each VM, create folder with its IP and copy log files with scp 
+    vms = ('10.0.0.?','10.0.0.?','10.0.0.?','10.0.0.?','10.0.0.?')
+    for ip in vms:
+        os.system('mkdir /tmp/oxa_log_files/'+ip)    
+        os.system('scp '+ip+':/edx/var/log/supervisor/*default* /tmp/oxa_log_files/'+ip+'/')
+
+	# In the copied log files search for failed activation emails and write to a temp file notsent	/tmp/oxa_log_files/notsentemails.txt
+    os.system("grep 'Unable to send activation email' /tmp/oxa_log_files/*/* > /tmp/oxa_log_files/notsentemails.txt")
+    
+	# Now read each line from /tmp/oxa_log_files/notsentemails.txt and if it is not processed already process it and insert to table and update statistics for the corersponding YYYY,MM,DD 
+	db = MySQLdb.connect(IP,User,Password,Database)
+    cursor = db.cursor()
+    with open('/tmp/oxa_log_files/notsentemails.txt') as f:
+        for line in f:
+            process_error_line(cursor,line)
```

Comment:
How many lines do we typically have to process this way? It seems if there are a lot, each one triggers a separate select only or select / update query.